src/types/$waveui.ts

```diff
@@ -10,10 +11,233 @@ export interface $waveui {
     md: boolean,
     lg: boolean,
     xl: boolean,
-    width: number|null
+    width: number|null,
   },
-  config: {},
-  colors: {}, // Object of pairs of color-name => color hex.
+  config: {
+    /**
+     * From version 3.0. Sets the Wave UI root on a custom node (expects a valid CSS selector)
+     * @property {string} on - Default: '#app', if not found body will be used
+     * @see https://antoniandre.github.io/wave-ui/options-presets-and-waveui
+     */
+    on: string,
+
+    /**
+     * Breakpoint settings
+     * @property breakpoints
+     * @see https://antoniandre.github.io/wave-ui/breakpoints#default-breakpoints
+     * @see https://antoniandre.github.io/wave-ui/options-presets-and-waveui
+     */
+    breakpoints: {
+      /**
+       * @property {number} xs - Default: 600
+       */
+      xs: number,
+      /**
+       * @property {number} sm - Default: 900
+       */
+      sm: number,
+      /**
+       * @property {number} md - Default: 1200
+       */
+      md: number,
+      /**
+       * @property {number} lg - Default: 1700
+       */
+      lg: number,
+      /**
+       * Xl only needs a greater value than lg but starts from lg and goes to infinity.
+       * @property {number} xl - Default: 9999
+       */
+      xl: number,
+    },
+
+    /**
+     * Css settings
+     * @property css
+     * @see https://antoniandre.github.io/wave-ui/colors
+     * @see https://antoniandre.github.io/wave-ui/options-presets-and-waveui
+     */
+    css: {
+      /**
+       * Generate shades for custom colors and status colors.
+       * NOTE: the color palette shades are always generated separately from SCSS.
+       * @property {boolean} colorShades - Default: true
+       */
+      colorShades: boolean,
+
+      /**
+       * Generate CSS variables for color shades.
+       * NOTE: the `colorShades` must be enabled for this to work.
+       * @property {boolean} colorShadeCssVariables - Default: false
+       */
+      colorShadeCssVariables: boolean,
```

Comment:
This it the actual new property for this PR. The rest of these settings are added into these types are from #145